lib/api/apiUtils/object/checkUserMetadataSize.js

```diff
@@ -0,0 +1,38 @@
+const { maximumMetaHeadersSize,
+    invalidObjectUserMetadataHeader } = require('../../../../constants');
+
+/**
+ * Checks the size of the user metadata in the object metadata and removes
+ * them from the response if the size of the user metadata is larger than
+ * the maximum size allowed. A custome metadata key is added to the response
+ * with the number of user metadata keys not returned as its value
+ * @param {object} objMD - object's metadata
+ * @return {object} responseMetaHeaders headers with object metadata to include
+ * in response to client
+ */
+function checkUserMetadataSize(objMD) {
+    let md = {};
+    let userMetadataSize = 0;
+    md = JSON.parse(JSON.stringify(objMD));
+    // collect the user metadata keys from the object metadata
+    const userMetadataHeaders = Object.keys(md)
+        .filter(key => key.startsWith('x-amz-meta-'));
+    // compute the size of all user metadata key and its value
+    userMetadataHeaders.forEach(header => {
+        userMetadataSize += header.length + md[header].length;
```

Comment:
Is the content stored internally UTF-8 encoded? I suppose the size could grow if it gets encoded down the line, before the response is sent. Just thinking out loud here—not sure about it.

Reply:
@bennettbuchanan from the official aws docs, https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html the acceptable encoding for user metadata is ascii/utf-8 and the 2KB restriction is for the same.